packages/one-app-dev-bundler/esbuild/plugins/generate-integrity-manifest.js

```diff
@@ -18,17 +18,18 @@ import ssri from 'ssri';
 import fs from 'node:fs';
 import { getJsFilenamesFromKeys } from '../utils/get-js-filenames-from-keys.js';
 
-async function writeIntegrityFragment(bundleName, integrityString, fileName) {
+// NOTE:: This needs to be synchronous to avoid race condition resulting in
+// missing integrity manifest entries
+function writeIntegrityFragment(bundleName, integrityString, fileName) {
   let integrityObject = {};
   try {
-    integrityObject = JSON.parse(await fs.promises.readFile(fileName, 'utf8'));
+    integrityObject = JSON.parse(fs.readFileSync(fileName, 'utf8'));
   } catch (e) {
     // empty catch in case file does not exist
   }
 
   integrityObject[bundleName] = integrityString;
-
-  return fs.promises.writeFile(fileName, JSON.stringify(integrityObject, null, 2));
+  return fs.writeFileSync(fileName, JSON.stringify(integrityObject, null, 2));
```

Comment:
Can all of this be put in the try-catch? If we fail to read the file, do we still want to be writing an empty object?

Reply:
if it fails this the error should bubble up and fail the build. The try catch squashes it